src/fmu/dataio/aggregation.py

```diff
@@ -317,13 +333,10 @@ def generate_metadata(
             uuids.append(xuuid)
 
         # first config file as template
-        self._generate_aggrd_metadata(obj, real_ids, uuids, compute_md5)
-        if skip_null:
-            self._metadata = _utils.drop_nones(self._metadata)
+        self._set_metadata(obj, real_ids, uuids, compute_md5)
```

Comment:
I think eventually we should move to call ```_set_metadata()``` in ```__post_init__()``` but that can be taken in a separate PR.